BookStore/validator/index.js

```diff
@@ -0,0 +1,15 @@
+exports.userSignupValidator = (req, res, next) => {
+    req.check("name", "Name is required!").notEmpty();
+    req.check("email")
+        .matches(/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/)
+        .withMessage("You must enter a valid EMAIL!");
+    req.check("password", "password is required!").notEmpty()
+        .matches(/^(?=.*[a-zA-Z])(?=.*[0-9])(?=.{8,})/)
+        .withMessage("password must contain at least 8 characters, a lower case char, an upper case char and a number!");
+    const errors = req.validationErrors();
+    if (errors) {
+        const firstError = errors.map(e => e.msg)[0];
+        return res.status(400).json({ error: firstError });
```

Comment:
I see you are using `res.status(400).json({ error: ... })` in a couple of places. 

Why don't you create an Express Error handling middleware for doing this:

```js
app.use((err, req, res, next) => {
  res.status(err.status).json({ error: err });
});
``` 

Then here, you set the status code on the error and call `next(err)`

```js
if (errors) {
        const firstError = errors[0];
        firstError.status = 400;
        // remember you must call next() with an Error object
        // I don't think it will work with a regular JS object, not 100% sure though
        next(firstError);
}
```

Reply:
we will work on it later

Reply:
Can you at least make an issue to do this so you don't forget?

Reply:
okay